### PR TITLE
fix(2024.6.0): Handling time zones without blocking the event loop

### PR DESCRIPTION
**Pull request by golles**
https://developers.home-assistant.io/blog/2024/05/19/fix_zoneinfo_blocking_io/

Until `2024.6.0` is released, this fix uses `2024.6.0b6` 
Fix tests for unloading custom component (https://github.com/home-assistant/core/pull/117214)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1 +1 @@
-homeassistant>=2023.9.0
+homeassistant>=2024.6.0b6
```
Comment (golles) [inline requirements.txt:1]: - [ ] After HA `2024.6.0` release update the version in this PR as well

**Comment by hmmbob** [comment]
If you're able to make this a beta release, I could test it for you

**Comment by golles** [comment]
Made a pre-release `2.2.0-b1` containing the fix.

I can't merge this yet, as https://github.com/MatthewFlamm/pytest-homeassistant-custom-component/releases has not released a compatible version for HA `2024.6.0` yet, it should be there tomorrow morning

**Comment by hmmbob** [comment]
Confirming that the beta fixes the issue in 2024.6.0

**Pull request merged**